image/image.go

```diff
@@ -787,3 +787,48 @@ func copyLocalSnapFile(snapPath, targetDir string, info *snap.Info) (dstPath str
 	dst := filepath.Join(targetDir, filepath.Base(info.MountFile()))
 	return dst, osutil.CopyFile(snapPath, dst, 0)
 }
+
+func ValidateSeed(seedFile string) error {
+	seed, err := snap.ReadSeedYaml(seedFile)
+	if err != nil {
+		return err
+	}
+
+	// read the snaps info
+	snapInfos := make(map[string]*snap.Info)
+	for _, seedSnap := range seed.Snaps {
+		fn := filepath.Join(filepath.Dir(seedFile), "snaps", seedSnap.File)
+		snapf, err := snap.Open(fn)
+		if err != nil {
+			return err
+		}
+		info, err := snap.ReadInfoFromSnapFile(snapf, nil)
+		if err != nil {
+			return err
+		}
+		snapInfos[info.InstanceName()] = info
+	}
+	// ensure we have either "core" or "snapd"
+	_, haveCore := snapInfos["core"]
+	_, haveSnapd := snapInfos["snapd"]
+	if !(haveCore || haveSnapd) {
+		return fmt.Errorf("the core or snapd snap must be part of the seed")
+	}
+
+	// check that all bases/default-providers are part of the seed
+	for _, info := range snapInfos {
+		if info.Base != "" && info.Base != "none" {
```

Comment:
if the snap is an app and name != "snapd",  Base == "" means we need to check that "core" is there